core/src/main/java/org/eclipse/hono/util/TenantConstants.java

```diff
@@ -0,0 +1,112 @@
+         * @return Action The Action as enum, or {@link Action#ACTION_UNKNOWN} otherwise.
+         */
+        public static Action from(final String subject) {
+            if (subject != null) {
+                try {
+                    return Action.valueOf(subject);
+                } catch (IllegalArgumentException e) {
+                }
+            }
+            return ACTION_UNKNOWN;
+        }
+
+        /**
+         * Helper method to check if a subject is a valid Tenant API action.
+         *
+         * @param subject The subject to validate.
+         * @return boolean {@link Boolean#TRUE} if the subject denotes a valid action, {@link Boolean#FALSE} otherwise.
+         */
+        public static boolean isValid(final String subject) {
+            return Action.from(subject) != Action.ACTION_UNKNOWN;
+        }
+    }
+
+    /* message payload fields */
+    public static final String FIELD_ADAPTERS                    = "adapters";
+    public static final String FIELD_ADAPTERS_TYPE               = "type";
+    public static final String FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED = "device-authentication-required";
+
+    public static final String FIELD_RESPONSE_STATUS = "status";
+
+    /**
+     * The name of the Tenant API endpoint.
+     */
+    public static final String TENANT_ENDPOINT = "tenant";
+
+    /**
+     * The vert.x event bus address to which inbound registration messages are published.
+     */
+    public static final String EVENT_BUS_ADDRESS_TENANT_IN = "tenant.in";
+
+    /**
+     * Creates a JSON object from a Tenant API request message.
+     *
+     * @param message The AMQP 1.0 tenant request message.
+     * @return The tenant message created from the AMQP message.
+     * @throws NullPointerException if message is {@code null}.
+     * @throws DecodeException if the message contains a body that cannot be parsed into a JSON object.
+     */
+    public static JsonObject getTenantMsg(final Message message) {
+        Objects.requireNonNull(message);
+        final String subject = message.getSubject();
+        final String tenantId = MessageHelper.getTenantIdAnnotation(message);
+        final JsonObject payload = MessageHelper.getJsonPayload(message);
+        return getServiceRequestAsJson(subject, tenantId, payload);
+    }
+
+    /**
+     * Gets a JSON object representing the reply to a Tenant API request via the vert.x event bus.
+     *
+     * @param tenantId The tenant for which the message was processed. If null, the value for the key
+     *                 {@link RequestResponseApiConstants#FIELD_TENANT_ID} in the reply will be null, too.
+     * @param tenantResult The result to return to the sender of the request.
+     * @return JsonObject The JSON reply object.
+     * @throws NullPointerException If tenantResult is null.
+     */
+    public static final JsonObject getServiceReplyAsJson(final String tenantId, final TenantResult tenantResult) {
```

Comment:
from a code style perspective it would be better to add an explicit check for tenantResult not being `null` here before starting to create new objects.

Reply:
:+1: